docs/src/example_networks/SPR_fitting.md

```diff
@@ -0,0 +1,105 @@
+# [Surface Plasmon Resonance Model Example](@id surface_plasmon_resonance_model_Example)
+This tutorial shows how to programmatically construct a ['ReactionSystem'](@ref) corresponding to the chemistry underlying the [Surface Plasmon Resonance Model](https://en.wikipedia.org/wiki/Surface_plasmon_resonance) using [ModelingToolkit](http://docs.sciml.ai/ModelingToolkit/stable/)/[Catalyst](http://docs.sciml.ai/Catalyst/stable/). You can find a simpler constructruction of this model [here](https://docs.sciml.ai/Catalyst/stable/catalyst_applications/parameter_estimation/) in the example of parameter estimation. 
+
+Using the symbolic interface, we will create our states/species, define our reactions, and add a discrete event. This model is simulated and used to generate sample data points with added noise and then used to fit parameters in iterations. This event will correspond to the time at which the antibody stop binding to antigen and switch to dissociating. 
+
+We begin by importing some necessary packages. 
+```julia
+using ModelingToolkit, Catalyst, DifferentialEquations
+using Plots, Optimization
+using OptimizatizationOptimJL, Optim
+```
+In this example, the concentration of antigen,`\beta` is varied to determine the constant of proportionality, `\alpha`, `k_{on}`(association rate constant), and `k_{off}` (dissociation rate constant) which characterized the binding interaction between the antigen and the immobilized receptor molecules on the sensor slide. We start by defining our reaction equations, parameters, variables, event, and states. 
+```julia
+@variables t 
+@parameters k_on k_off α
+@species A(t) B(t)
+
+rxs = [(@reaction α*k_on, A --> B), (@reaction k_off, B --> A)]
```

Comment:
Is there a reason that we don't use the DSL here? Simply doing
```julia
rs = @reaction_network begin
    (α*k_on, k_off), A <--> B
end
```
seems a lot more intuitive? The main difference is that we cannot add events in the DSL, but we could just add this as a PresettimeCallback: https://docs.sciml.ai/Catalyst/stable/catalyst_applications/advanced_simulations/#advanced_simulations_callbacks

Reply:
The idea is to use a symbolic callback so the callback is encoded in the Catalyst model. Otherwise if one uses the DSL one must use `extend` to merge a separate system with the event in to the Catalyst reaction model.

Once we get discrete event support in the DSL we can rewrite this to fully use the DSL.